packages/@aws-cdk/aws-gamelift/test/queued-matchmaking-configuration.test.ts

```diff
@@ -0,0 +1,534 @@
+import { Template } from '@aws-cdk/assertions';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as cdk from '@aws-cdk/core';
+import { Duration } from '@aws-cdk/core';
+import * as gamelift from '../lib';
+
+describe('queuedMatchmakingConfiguration', () => {
+
+  describe('new', () => {
+    let stack: cdk.Stack;
+    const ruleSetBody = JSON.stringify('{}');
+    let gameSessionQueue: gamelift.GameSessionQueueBase;
+    let ruleSet: gamelift.MatchmakingRuleSet;
+
+    beforeEach(() => {
+      stack = new cdk.Stack();
+      gameSessionQueue = new gamelift.GameSessionQueue(stack, 'MyGameSessionQueue', {
+        gameSessionQueueName: 'test-queue-name',
+        destinations: [],
+      });
+      ruleSet = new gamelift.MatchmakingRuleSet(stack, 'MyMatchmakingRuleSet', {
+        matchmakingRuleSetName: 'test-ruleSet',
+        content: gamelift.RuleSetContent.fromInline(ruleSetBody),
+      });
+    });
+
+    test('default QueuedMatchmakingConfiguration', () => {
+      new gamelift.QueuedMatchmakingConfiguration(stack, 'MyQueuedMatchmakingConfiguration', {
+        matchmakingConfigurationName: 'test-config-name',
+        gameSessionQueues: [gameSessionQueue],
+        ruleSet: ruleSet,
+      });
+
+      Template.fromStack(stack).hasResource('AWS::GameLift::MatchmakingConfiguration', {
+        Properties:
+            {
+              Name: 'test-config-name',
+              NotificationTarget: { Ref: 'MyQueuedMatchmakingConfigurationTopicBAC3E679' },
+              FlexMatchMode: 'WITH_QUEUE',
+              BackfillMode: 'AUTOMATIC',
+              AcceptanceRequired: false,
+              RequestTimeoutSeconds: 300,
+              RuleSetName: { Ref: 'MyMatchmakingRuleSet41F295C4' },
+              GameSessionQueueArns: [
+                {
+                  'Fn::Join': [
+                    '',
+                    [
+                      'arn:',
+                      {
+                        Ref: 'AWS::Partition',
+                      },
+                      ':gamelift:',
+                      {
+                        Ref: 'AWS::Region',
+                      },
+                      ':',
+                      {
+                        Ref: 'AWS::AccountId',
+                      },
+                      ':gamesessionqueue/',
+                      {
+                        Ref: 'MyGameSessionQueue1A15CE31',
+                      },
+                    ],
+                  ],
+                },
+              ],
+            },
+      });
```

Comment:
Something is wrong with this formatting, but I'm not sure what.

Reply:
formatting fixed